### DISCUSSION TITLE
Client enquiery regarding HC30PL / MoveIt interoperability

### DISCUSSION THREAD

**Discussion by charleswilmot** [Yaskawa-Global/motoros2#75] [Q&A]
At my company (Festo), we are currently deciding whether we should order Yaskawa MOTOMAN HC30PL arms or Fanuc CRX20IA arms.

Being able to use MoveIt is a necessary condition for us to purchase the Yaskawa robots.

Is there someone that we could contact before we take a decision? We would need to have one robot ready to use with MoveIt before October 2023.

_I initially posted this question in the issue ticket #74 , I deleted the original post and reposted here, following indications from @gavanderhoorn  _

**Comment by ted-miller** [comment]
Hi @charleswilmot, we at Yaskawa do not have sufficient experience with MoveIt2 to comment on how to create that package at this time. These packages have been created by the ROS community in the past.

There are detailed tutorials available at https://moveit.picknik.ai/main/index.html. Additionally, they may be able to provide assistance on the MoveIt2 discussion boards https://github.com/orgs/ros-planning/discussions.

However, we can certainly provide the supporting robot definition files ([like these](https://github.com/gavanderhoorn/motoman/tree/ros2)) for an HC30PL robot arm. This is a newer arm and we have not yet created the support package for it.
@EricMarcil: Can you assist with this? (Note that the ROS2 packages in that link are slightly different than their corresponding ROS1 files. Also, there is a [note here](https://github.com/Yaskawa-Global/motoros2/issues/73#issue-1766836987) that mentions the change in the joint naming-convention.)

**Reply by gavanderhoorn** [reply]
@EricMarcil: I'd suggest creating the ROS 1 support package first. I can do the conversion to ROS 2.

**Comment by gavanderhoorn** [comment]
@charleswilmot: could you clarify whether you're looking to use ROS 1 with MoveIt, or ROS 2 with MoveIt 2?

**Comment by charleswilmot** [comment]
Thanks much @ted-miller ,

Already pleased to get a quick answer!

I will try to asses more precisely how much work the HC30PL / MoveIt2 integration involves given the information that you gave us.

> could you clarify whether you're looking to use ROS 1 with MoveIt, or ROS 2 with MoveIt 2?

I am not well informed regarding the existing interfaces compatible with the Yaskawa robots. We intend to control the robot from python, and my current understanding is that the preferred approach would be to use ROS2 for that purpose. Therefore we would use ROS2 + MoveIt 2.

However, the new robots would come as a replacement for old robots that were relying on ROS1 + MoveIt.
